### PR TITLE
Zod fix union error + reduce resolver size

### PR DESCRIPTION
**Pull request by jorisre**
- Fix Zod resolver union error (same as https://github.com/react-hook-form/resolvers/pull/119)
- Extract test fixtures
- Reduce zod resolver size

### CODE REVIEW

**Comment by bluebill1049** [comment]
oh wow!!!

File: zod/src/zod.ts [hunk 1]
```diff
@@ -1,67 +1,64 @@
...
 const parseErrorSchema = (
-  zodError: z.ZodError,
+  zodErrors: z.ZodSuberror[],
   validateAllFieldCriteria: boolean,
 ) => {
-  if (zodError.isEmpty) {
-    return {};
-  }
+  const errors: Record<string, FieldError> = {};
+  for (; zodErrors.length; ) {
```
Comment (bluebill1049) [inline zod/src/zod.ts:11]: didn't know you can do this!

**Review by bluebill1049 — APPROVED** [review]
awesome!

File: zod/src/zod.ts (hunk 1 above)
Comment (jorisre) [reply zod/src/zod.ts:11]: Saving bytes 😬  More coming ;)

**Pull request merged**